### DISCUSSION TITLE
Feature Request: Importing Across Platforms

### DISCUSSION THREAD

**Discussion by michaeljmcd** [MikeMaximus/gbm#253] [Ideas]
I currently have Game Backup Monitor installed on two machines, my gaming machine that runs Windows and my laptop which runs Linux. Both backup to separate folders in OneDrive (i.e. '$HOME/OneDrive/Game_Backups/${HOSTNAME}`). One thing I like to do for less hardware intensive games is to switch back and forth between playing on one and then playing on the other.

This setup is possible, but a little awkward. When I switch machines, I need to do a restore (as noted in http://mikemaximus.github.io/gbm-web/linux.html) and then fix the game save path which gets reset to the path on the other operating system.

I can think of a couple of related ways to make this less painful. One is to allow backup data to be restored without also restoring the metadata from ` _gbm_backup_metadata.xml`. Some of the clunkiness would be there, but at least the paths would be ironed out.

Another nice-to-have would be for there to be a hook to automatically copy in the latest backup from another GBM setup. It would probably need the feature above to avoid the path overwriting.

Either way, the core use case is playing the same game (by turns) on different machines running different OSes.

**Comment by MikeMaximus** [comment]
Thanks for bringing this to my attention.  It's an oversight, I haven't done any gaming on Linux for a few years or I'd have run into this myself after implementing the metadata.  I'll get a solution into the next release.

**Comment by MikeMaximus** [comment]
I started looking into this today because I couldn't remember exactly how I designed the cross OS import.

It isn't well documented unless you plow through the whole manual, but there are two methods of importing backup files.  One is accessed from "Tools -> Import Backup" and the other is available inside the Game Manager in the "Backup Management" tab when a specific game is selected.

![image](https://user-images.githubusercontent.com/15719732/137209034-d16b6d92-c0be-4b88-867c-ce67be0b9cec.png)

The Game Manager import method the one you want to be using when importing a backup from a different OS, because it will allow you to force the import of any backup file and ignore the metadata.

1.  Select the Linux version of the game configuration.
2.  Use the Import Backup option in the Backup Management tab.
3.  Choose the Windows backup file and click "Yes" to the warning message that appears.

![image](https://user-images.githubusercontent.com/15719732/137209419-e68fa146-339b-4dee-8ce8-894bf87f0452.png)

This will allow you to import backups from Windows without messing around with the configuration every single time.  I know it's still janky and I'll see if I can think of a better way for the future.  But I hope this helps you out for now.

**Comment by michaeljmcd** [comment]
Importing through the Game Manager does indeed work. Thanks for the workaround!
